scripts/cc-cli/src/test/integration-tests/collectCoins.test.ts

```diff
@@ -0,0 +1,142 @@
+import { parseAmountInternal } from "../../utils/parsing";
+import { signSendAndWatch } from "../../utils/tx";
+import { fundFromSudo, randomTestAccount } from "./helpers";
+import {
+  arg,
+  deployCtcToken,
+  retry,
+  setArg,
+  setArgIfUndef,
+  setAuthorities,
+} from "../../globalSetup";
+import execa from "execa";
+import {
+  Blockchain,
+  CreditcoinApi,
+  KeyringPair,
+  Wallet,
+  creditcoinApi,
+  providers,
+} from "creditcoin-js";
+import { cryptoWaitReady } from "@polkadot/util-crypto";
+import { testData, tryRegisterAddress } from "creditcoin-js/lib/testUtils";
+import { describeIf } from "../../utils/tests";
+import { getBalance } from "../../utils/balance";
+
+describeIf(arg("CREDITCOIN_EXECUTE_SETUP_AUTHORITY"), "collect-coins", () => {
+  let ccApi: CreditcoinApi;
+  let sudo: KeyringPair;
+  let caller: any;
+
+  const { keyring, blockchain } = testData(
+    arg("CREDITCOIN_ETHEREUM_CHAIN") as Blockchain,
+    arg("CREDITCOIN_CREATE_WALLET"),
+  );
+
+  beforeAll(async () => {
+    await deployCtcToken((global as any).CREDITCOIN_CTC_CONTRACT_ADDRESS);
+    setArg(
+      "CREDITCOIN_CTC_CONTRACT_ADDRESS",
+      process.env.CREDITCOIN_CTC_CONTRACT_ADDRESS,
+    );
+    setArgIfUndef(
+      "CREDITCOIN_CTC_BURN_TX_HASH",
+      process.env.CREDITCOIN_CTC_BURN_TX_HASH,
+    );
+
+    await retry(5, async (resolve: any, reject: any) => {
+      await setAuthorities().then(resolve).catch(reject);
+    }).catch(() => {
```

Comment:
```suggestion
    await retry(5, setAuthorities).catch(() => {
```

If you choose to accept my suggestion for the retry function, on the caller side it would just look like this